packages/components/src/utils/unit-values.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Parses a number and unit from a value.
+ *
+ * @param  toParse Value to parse
+ *
+ * @return  The extracted number and unit.
+ */
+export function parseUnitValue(
+	toParse: string | number
+): [ number | string | undefined, string | undefined ] {
+	// eslint-disable-next-line eqeqeq
+	if ( toParse == null ) {
+		return [ undefined, undefined ];
+	}
+
+	const value = String( toParse ).trim();
+
+	let num: number | string = parseFloat( value );
+	num = Number.isNaN( num ) ? '' : num;
```

Comment:
Would this be a good place to use [`isValueNumeric`](https://github.com/WordPress/gutenberg/pull/33206)?

Reply:
No because CSS numbers can only be formatted the same way parseFloat works (no comma separators, decimals separated by periods, etc).